docker-compose.yaml

```diff
@@ -51,10 +53,48 @@ services:
     environment:
       - SPARK_MASTER_URL=spark://spark-master:7077
 
+  minio:
+    image: minio/minio
+    container_name: spark-minio
+    expose:
+      - "9000"
+    ports:
+      - "9000:9000"
+      # MinIO Console is available at http://localhost:9001
+      - "9001:9001"
+    environment:
+      MINIO_ROOT_USER: minio
+      MINIO_ROOT_PASSWORD: minio123
+      MINIO_ACCESS_KEY: minio
+      MINIO_SECRET_KEY: minio123
+    healthcheck:
+      test: timeout 5s bash -c ':> /dev/tcp/127.0.0.1/9000' || exit 1
+      interval: 1s
+      timeout: 10s
+      retries: 5
+    # Note there is no bucket by default
+    command: server /data --console-address ":9001"
+
+  minio-create-bucket:
+    image: minio/mc
+    depends_on:
+      minio:
+        condition: service_healthy
+    entrypoint: >
+      bash -c "
+      mc alias set minio http://minio:9000 minio minio123 &&
+      if ! mc ls minio/delta-lake 2>/dev/null; then
+        mc mb minio/delta-lake && echo 'Bucket delta-lake created'
+      else
+        echo 'bucket delta-lake already exists'
+      fi
+      "
+
   notebook:
     build:
       context: .
       dockerfile: Dockerfile
+    entrypoint: /opt/scripts/notebook_entrypoint.sh
```

Comment:
This means we're going to have to configure the entrypoint in rancher, which devops is generally trying to avoid

Reply:
yea. I configured this in Rancher. I cannot think of other ways to use this as both master and notebook.

Reply:
Have an entrypoint that takes an env var that selects the correct mode

Reply:
We don't really know what the base image does for different mode.

Reply:
I'm not sure what you mean, the base image is pretty well documented

Reply:
👍 but again, not worth it and might cause problem if base image is updated to use a different entrypoint script. Way easier to maintain by just giving Rancher the entrypoint script.

Reply:
Not for the people setting up the images. If the base image entrypoint changes we just change ours

Reply:
If there's an ENTRYPOINT there's no need for CMD, is there?

Reply:
Actually I think the current setup will do something weird, as the contents of CMD will be provided as arguments to ENTRYPOINT IIUC

Reply:
I tried to put `/opt/bitnami/scripts/spark/run.sh` at the end of main entrypoint.sh file. The master services failed to start.

Reply:
so in `/opt/bitnami/scripts/spark/entrypoint.sh`

```
if [[ "$1" = "/opt/bitnami/scripts/spark/run.sh" ]]; then
    info "** Starting Spark setup **"
    /opt/bitnami/scripts/spark/setup.sh
    info "** Spark setup finished! **"
fi
```

Reply:
removed `run.sh` to the entypoint as arguments and added notes to explain it.

Reply:
:+1: